### PR TITLE
"Feature gate" crates, un-default non-Rust crates.

### PR DESCRIPTION
**Pull request by DanielKeep**
This makes every crate optional, with a nearly full set enabled by default.  The crates that are *not* enabled by default are those that do not build cleanly on Windows without additional configuration.  There is also an "all" feature that pulls in everything.

The README now contains the following:

> They might be disabled because they do not work well on all platforms, because they are very large, or because they require special environment configuration to build.

I propose that this be the general position taken with stdx: if a crate does not build on all platforms, is very large (which would disincentivise people from using stdx on smaller projects), or requires additional external configuration (downloading compilers, headers, libraries, changing environment variables, etc.), then it should *not* be in the default set.  This would include all crates that depend on non-Rust, non-ubiquitous libraries (*i.e.* things like POSIX on *nix systems, or Win32 on Windows, are OK.)

In other words: if it won't build on a fresh OS install that has *just* had Rust installed, it probably shouldn't be default.

I've gated hyper on the basis that it depends on openssl.  I have already gone through the pain of working out how to set up my machine to build this... except now I find it doesn't work any more and I have no idea *why*.  I *have* MinGW and the OpenSSL headers installed, they just... stopped working for some reason.

I've gated flate2 on the basis of Tomaka's complaints in the Discourse thread discussion of stdx.  It builds for me, but then, I have a non-default configuration.

I have *not* gated time, as the problem there appears to be depending on a very old version of the crate; I've just changed the version number from 0.1.3 to 0.1.30.

### CODE REVIEW

**Comment by brson** [comment]
This is perfect, thanks.
